glib-macros/tests/properties.rs

```diff
@@ -101,10 +120,16 @@ mod foo {
             double: RefCell<f64>,
             #[property(get = |_| 42.0, set)]
             infer_inline_type: RefCell<f64>,
+            #[property(get, set)]
+            custom_type: RefCell<MyInt>,
             // The following property doesn't store any data. The value of the property is calculated
             // when the value is accessed.
             #[property(get = Self::hello_world)]
             _buzz: PhantomData<String>,
+            #[property(get, set)]
+            thread_guard_wrapped: Mutex<ThreadGuard<u32>>,
+            #[property(get, set)]
+            thread_guard_wrapping: ThreadGuard<Mutex<u32>>,
```

Comment:
Would this also already handle a `Mutex<Option<ThreadGuard<Paintable>>>`? Asking for the `Option` there. It's a write-only property so I think that potentially works because there's `From<T> for Option<T>` but I think it doesn't work in practice because you'd need to call `from()` twice?